src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,253 @@
+package frc.robot.subsystems.conveyor;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.wpilibj.*;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Deque;
+
+import static frc.robot.Ports.Conveyor.MOTOR_INVERSION;
+import static frc.robot.Ports.Conveyor.SOLENOID;
+
+public class Conveyor extends SubsystemBase {
+    private static Conveyor INSTANCE = null;
+    private final WPI_TalonFX motor = new WPI_TalonFX(Ports.Conveyor.MOTOR);
+    private final Deque<String> cargoPositions = new ArrayDeque<>();
+    private final ColorSensorV3 colorSensor = new ColorSensorV3(I2C.Port.kMXP);
+    private final Solenoid flap = new Solenoid(PneumaticsModuleType.CTREPCM, SOLENOID);
+    private final DigitalInput postFlapBeam = new DigitalInput(Ports.Conveyor.POST_FLAP_BEAM_BREAKER);
+    private final DigitalInput preFlapBeam = new DigitalInput(Ports.Conveyor.PRE_FLAP_BEAM_BREAKER);
+    private final ColorMatch match = new ColorMatch();
+    private DriverStation.Alliance lastSeenColor = DriverStation.Alliance.Invalid;
+    private boolean wasPostFlapBeamActive = true;
+
+    private Conveyor() {
+        motor.setInverted(MOTOR_INVERSION);
+        match.addColorMatch(Constants.Conveyor.RED);
+        match.addColorMatch(Constants.Conveyor.BLUE);
+        match.addColorMatch(Constants.Conveyor.NONE);
+        cargoPositions.add(DriverStation.Alliance.Invalid.name());
+        cargoPositions.add(DriverStation.Alliance.Invalid.name());
+    }
+
+    /**
+     * lazy instantiation
+     *
+     * @return the subsystem instance
+     */
+    public static Conveyor getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Conveyor();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * @return the color sensor value as a DriverStation.Alliance enum.
+     */
+    public DriverStation.Alliance getColor() {
+        Color color = colorSensor.getColor();
+        ColorMatchResult result = match.matchClosestColor(color);
+        Color resultColor = result.color;
+        SmartDashboard.putBoolean("match", resultColor == Color.kRed);
+
+        if (resultColor == Constants.Conveyor.RED) {
+            return DriverStation.Alliance.Red;
+        } else if (resultColor == Constants.Conveyor.BLUE) {
+            return DriverStation.Alliance.Blue;
+        } else {
+            return DriverStation.Alliance.Invalid;
+        }

```

Comment:
It's you choice but you can write it like this:
```suggestion
        if (resultColor == Constants.Conveyor.RED) {
            return DriverStation.Alliance.Red;
        } 
        if (resultColor == Constants.Conveyor.BLUE) {
            return DriverStation.Alliance.Blue;
        }
        return DriverStation.Alliance.Invalid;
```